### PR TITLE
Mitre Attack SP Update

### PR DESCRIPTION
**Pull request by cs-rohit-khune**
a. Added Mock Output in all the Hunt MITRE Attack Playbooks
b. Added new playbooks to link 'Techniques' and 'Sub-Techniques' on update of MITRE ATTACK ID field
c. Updated logic to fetch comments corelated to hunt record in playbook " 13 - MITRE ATT&CK™ - Modulars > Deduplicate Comments (Hunt)"

UT Steps:
  1. Execute playbook from Hunt record for specific MITRE Attack
  2. Choose SIEM from manual decision pop-up e.g Elastic
  3. Validate if alert is created for MITRE Attack ID
  4. Validate if alert is linked to corresponding MITRE Attack technique or sub-technique
  5. Validate if comments relavent to alert has been added in Hunt record
  6. Validate if duplicate comments has been deleted by comment deduplication playbook

### CODE REVIEW

**Review by cs-priyank-mankad — APPROVED** [review]
LGTM

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json [hunk 1]
```diff
@@ -137,18 +137,19 @@
                     "query": "@timestamp:{ {{vars.elasticHuntStart}} TO {{vars.elasticHuntEnd}} } AND (event_id:\"4765\" OR event_id: \"4766\")",
                     "run_as_user": ""
                 },
-                "version": "2.2.0",
+                "version": "2.2.1",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json:140]: Why we changed the connector version here? Have we tested with this connector?

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json [hunk 2]
```diff
@@ -5,7 +5,7 @@
     "aliasName": null,
     "tag": "ATT&CK, SID-History_Injection, Privilege_Escalation",
     "description": "Hunts for SID-History injection via mimikatz and other tools. Also hunts for SID-History added to accounts (success and failure). Adding SID-History may allow for escalated privileges if SID filtering is not enabled.",
-    "isActive": false,
+    "isActive": true,
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json:8]: Why it was deactivate earlier and activated now?

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json [hunk 3]
```diff
@@ -599,183 +601,184 @@
...
...      \"ParentCommandLine\": \"\\\"cmd.exe\\\" \/s \/k pushd \\\"C:\\\\Users\\\\superman\\\\Desktop\\\"\",\n              \"ParentProcessGuid\": \"{DA929735-D65B-5D1C-0000-001067EA1A00}\",\n              \"TerminalSessionId\": \"1\"\n            },\n            \"event_type\": \"sysmon\",\n            \"image_path\": \"C:\\\\Users\\\\superman\\\\Desktop\\\\invoice-156784.scr\",\n            \"process_id\": 1424,\n            \"source_name\": \"Microsoft-Windows-Sysmon\",\n            \"computer_name\": \"win7-host1-PC.cybercom.local\",\n            \"logstash_time\": 0.005793094635009766,\n            \"provider_guid\": \"{5770385F-C22A-43E0-BF4C-06F5698FFBD9}\",\n            \"record_number\": \"1985149\",\n            \"parent_image_path\": \"C:\\\\Windows\\\\System32\\\\cmd.exe\"\n          }\n        }\n      ],\n      \"total\": {\n        \"value\": 2,\n        \"relation\": \"eq\"\n      },\n      \"max_score\": 2.575364\n    },\n    \"took\": 996,\n    \"_shards\": {\n      \"total\": 95,\n      \"failed\": 0,\n      \"skipped\": 0,\n      \"successful\": 95\n    },\n    \"timed_out\": false\n  },\n  \"status\": \"Success\",\n  \"message\": \"\",\n  \"operation\": null\n}",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json]: Just want to confirm that, it has does not contain any customer data or customer specific details?

File: playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json [hunk 4]
```diff
@@ -5,16 +5,16 @@
     "aliasName": null,
     "tag": "ATT&CK, Persistence, Winlogon_Helper_DLL",
     "description": "Hunts for abnormal DLL loads and processes spawned by Winlogon",
-    "isActive": false,
+    "isActive": true,
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json:8]: same as above

File: playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json [hunk 5]
```diff
@@ -235,18 +236,19 @@
...
...,\n              \"CurrentDirectory\": \"C:\\\\Windows\\\\system32\\\\\",\n              \"ParentCommandLine\": \"winlogon.exe\",\n              \"ParentProcessGuid\": \"{BAECEEB0-779B-5D1A-0000-00100AAF0000}\",\n              \"TerminalSessionId\": \"1\"\n            },\n            \"event_type\": \"sysmon\",\n            \"image_path\": \"C:\\\\MaybeNotACredentialDumper.EXE\",\n            \"process_id\": 1460,\n            \"source_name\": \"Microsoft-Windows-Sysmon\",\n            \"computer_name\": \"admin-PC2.cybercom.local\",\n            \"logstash_time\": 0.004251003265380859,\n            \"provider_guid\": \"{5770385F-C22A-43E0-BF4C-06F5698FFBD9}\",\n            \"record_number\": \"187156\",\n            \"parent_image_path\": \"C:\\\\Windows\\\\System32\\\\winlogon.exe\"\n          }\n        }\n      ],\n      \"total\": {\n        \"value\": 1,\n        \"relation\": \"eq\"\n      },\n      \"max_score\": 3.33305\n    },\n    \"took\": 62,\n    \"_shards\": {\n      \"total\": 95,\n      \"failed\": 0,\n      \"skipped\": 0,\n      \"successful\": 95\n    },\n    \"timed_out\": false\n  },\n  \"status\": \"Success\",\n  \"message\": \"\",\n  \"operation\": null\n}",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json]: same as above

File: playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003) Part2.json [hunk 6]
```diff
@@ -33,30 +33,17 @@
             "arguments": {
                 "step_variables": {
                     "input": {
-                        "params": {
-                            "huntIRI": "{{ vars.huntIRI }}",
-                            "hostname": "{{ vars.hostname }}",
-                            "siemUsed": "{{ vars.siemUsed }}",
-                            "epochTime": "{{ vars.epochTime }}",
-                            "processID": "{{ vars.processID }}",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003) Part2.json:41]: Why we removed this?

File: playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003).json [hunk 7]
```diff
@@ -5,16 +5,16 @@
     "aliasName": null,
     "tag": "ATT&CK, Credential_Dumping, Credential_Access",
     "description": "Hunts for non-Windows processes accessing the lsass.exe process, which can be indicative of credential dumping",
-    "isActive": false,
+    "isActive": true,
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003).json:8]: same as above

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json [hunk 8]
```diff
@@ -1,20 +1,20 @@
 {
     "@type": "Workflow",
     "triggerLimit": null,
-    "name": "Link ATT&CK technique to Alert",
+    "name": "Link ATT&CK technique to Alert (On Create)",
     "aliasName": null,
     "tag": null,
-    "description": null,
-    "isActive": false,
+    "description": "Links MITRE technique or sub-technique to Alert, based on MITRE Attack ID",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json]: to an alert,

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json [hunk 9]
```diff
@@ -1,20 +1,20 @@
...
+    "isActive": true,
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json:8]: same as above? why activated now?

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Incident (On Update).json [hunk 10]
```diff
@@ -0,0 +1,260 @@
+{
+    "@type": "Workflow",
+    "triggerLimit": null,
+    "name": "Link ATT&CK technique to Incident (On Update)",
+    "aliasName": null,
+    "tag": null,
+    "description": "Links MITRE technique or sub-technique to Incident, based on MITRE Attack ID",
```
Comment (cs-suyog-jinturkar) [inline playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Incident (On Update).json]: to an incident

**Review by cs-suyog-jinturkar — CHANGES_REQUESTED** [review]
Please look at the review comments.

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json (hunk 1 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json:140]: This seems to be changed automatically, the old playbooks were built using v2.2.0 and the latest Mitre attack SP installs Elasticsearch v2.2.1 connector. Yes, all the testing has been done v2.2.1

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json (hunk 2 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json:8]: These playbooks are manually triggered on Hunt record, in order to make it visible in the "Execute" button, hence activating the playbook.

File: playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json (hunk 4 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json:8]: These playbooks are manually triggered on Hunt record, in order to make it visible in the "Execute" button, hence activating the playbook.

File: playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003).json (hunk 7 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003).json:8]: These playbooks are manually triggered on Hunt record, in order to make it visible in the "Execute" button, hence activated the playbooks.

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json (hunk 9 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json:8]: This is the 'Post Create' triggered on the alert record, hence activating the playbook. 
Note: The trigger condition checks if the MITRE ATT&CK ID field is populated or not. So this playbook triggers only mitre related alerts

File: playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003) Part2.json (hunk 6 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Credential Access/HUNTS - OS Credential Dumping (T1003) Part2.json:41]: This was automatically removed, I am investigating this. However, this has no impact on playbook operation.

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json (hunk 8 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Alert.json]: I have changed the description to "Links MITRE technique or sub-technique to an alert, based on MITRE Attack ID"

File: playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Incident (On Update).json (hunk 10 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Link Techniques to Alerts and Incidents/Link ATT&CK technique to Incident (On Update).json]: updated playbook description to "Links MITRE technique or sub-technique to an incident, based on MITRE Attack ID"

File: playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json (hunk 5 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Boot or Logon Autostart Execution/HUNTS - Winlogon Helper DLL (T1547.004).json]: found some references for Cybercom, updated the same from all the playbooks by removing the references

File: playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json (hunk 3 above)
Comment (cs-rohit-khune) [reply playbooks/13 - MITRE ATT&CK - Access Token Manipulation/HUNTS - SID-History Injection (T1134.005).json]: found some references for Cybercom, updated the same from all the playbooks by removing the references

**Pull request merged**